services/visual-identity/api.go

```diff
@@ -0,0 +1,75 @@
+package visualIdentity
+
+import (
+	"bufio"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	// 20bytes of data described by 14 emojis requires at least 2757 length alphabet
+	alphabetLen = 2757
+	hashLen     = 14
+)
+
+func NewAPI() *API {
+	emojis, _ := loadAlphabet(alphabetLen)
```

Comment:
This could fail silently. What do you think if instead loading the alphabet is done in the service
```go
type Service struct {
     api *API
}

func NewService() *Service {
	return &Service{
            api: NewAPI(),
        }
}

func (s *Service) Init() error {
     alphabet, err := loadAlphabet(alphabetLen)
     if err == nil {
         s.alphabet = alphabet
     }
    return err
}

func (s *Service) APIs() []rpc.API {
	return []rpc.API{
		{
			Namespace: "visualIdentity",
			Version:   "0.1.0",
			Service:   s.api,
			Public:    true,
		},
	}
}

```

And then in node/status_node_services.go initing the service could be created like this
```go
visualIdentityService, err := b.visualIdentityService()
if err != nil {
	return err
}
services = append(services, visualIdentityService)

...
...

func (b *StatusNode) visualIdentityService() (*gif.Service, error) {
        if b.visualIdentitySrvc != nil {
               return b.visualIdentitySrvc, nil
        }

	serv, err := visualIdentity.NewService()
        if err != nil {
             return nil, err
        }

        b.visualIdentitySrvc = serv
        return b.visualIdentitySrvc, nil
}
```